core/domain/exp_jobs_one_off_test.py

```diff
@@ -92,6 +92,216 @@ def run_job_for_deleted_exp(
         self.assertEqual(job_class.get_output(job_id), [])
 
 
+class MockExplorationModelWithDeprecatedFields(exp_models.ExplorationModel):
+    """Mock ExplorationModel to be able to set skill_tags, default_skin,
+    and skin_customizations.
+    """
+
+    skill_tags = (
+        datastore_services.StringProperty(indexed=True, repeated=True))
+    default_skin = (
+        datastore_services.StringProperty(default='conversation_v1'))
+    skin_customizations = datastore_services.JsonProperty(indexed=False)
+
+    def _trusted_commit(
+            self, committer_id, commit_type, commit_message, commit_cmds):
+        """Override this to escape recursion, which will otherwise occur
+        when super() is used in a mocked class.
+        """
+
+        base_models.VersionedModel._trusted_commit( # pylint: disable=protected-access
+            self, committer_id, commit_type, commit_message, commit_cmds)
+        exp_rights = exp_models.ExplorationRightsModel.get_by_id(self.id)
+
+        # TODO(msl): Test if put_async() leads to any problems (make
```

Comment:
TODOs should reference issue numbers. I'm not really sure you need one here, though.

Reply:
This was a code snippet that I took directly from the base `VersionedModel` class, so I'm not too sure. I didn't want to exclude it in case the TODO done over there also needed to be changed here.

Reply:
Yep, the snippet is not needed here.

Reply:
Can you check with @vojtechjelinek on this? I know he is investigating the put_async() issue with @rohitkatlaa, though I'm not sure if you need it in this test file.

Btw @vojtechjelinek @rohitkatlaa It might be worth redirecting that TODO to an issue number when you get a chance.

Reply:
Ok, I think I can take out all the code to do with the commit log since we're just mocking the commit anyway.